service/history/workflow/mutable_state_impl.go

```diff
@@ -2074,7 +2074,6 @@ func (e *MutableStateImpl) ReplicateActivityTaskStartedEvent(
 	ai.StartedEventId = event.GetEventId()
 	ai.RequestId = attributes.GetRequestId()
 	ai.StartedTime = event.GetEventTime()
-	ai.LastHeartbeatUpdateTime = ai.StartedTime
```

Comment:
There's another one in `AddActivityTaskStartedEvent`.

Also, we need to be very careful on this. and make sure it won't break sync activity task & heartbeat timeout calculation. Have you checked all usage of this field?

Reply:
I don't see any places relying on this being non-null in the value-read usages. It looks like they all use timestamp.TimeValue which has null checks. In DescribeWorkflowExecution, we are already only optionally setting that field. In timer_sequence:getActivityHeartbeatTimeout, we default to the start time if the LastHeartbeatUpdateTime is zero, which will have the same behavior. We also replace the field, but I don't see why that would cause any issues because the way it's read will still be the same usages as the ones I already checked.

Reply:
Also, thanks for catching the one in AddActivityTaskStartedEvent, fixed.